src/openlcb/MemoryConfigClient.hxx

```diff
@@ -270,7 +296,156 @@ private:
         cleanup_read();
         return return_ok();
     }
-    
+
+    Action do_write()
+    {
+        dgClient_ = full_allocation_result(dg_service()->client_allocator());
+        offset_ = request()->address;
+        payloadOffset_ = 0;
+        memoryConfigHandler_->set_client(&responseFlow_);
+        return call_immediately(STATE(send_next_write));
+    }
+
+    Action send_next_write()
+    {
+        return allocate_and_call(
+            dg_service()->iface()->dispatcher(), STATE(send_write_datagram));
+    }
+
+    Action send_write_datagram()
+    {
+        auto *b = get_allocation_result(dg_service()->iface()->dispatcher());
+        b->set_done(bn_.reset(this));
+        unsigned sz = request()->payload.size() - payloadOffset_;
+        if (sz > 64)
+            sz = 64;
+        writeLength_ = sz;
+        b->data()->reset(Defs::MTI_DATAGRAM, node_->node_id(), request()->dst,
+            MemoryConfigDefs::write_datagram(request()->memory_space, offset_,
+                             request()->payload.substr(payloadOffset_, sz)));
+        isWaitingForTimer_ = 0;
+        responseCode_ = DatagramClient::OPERATION_PENDING;
+        dgClient_->write_datagram(b);
+        return wait_and_call(STATE(write_complete));
+    }
+
+    Action write_complete()
+    {
+        if (!(dgClient_->result() & DatagramClient::OPERATION_SUCCESS))
+        {
+            // some error occurred.
+            return handle_write_error(dgClient_->result());
+        }
+        if (responseCode_ & DatagramClient::OPERATION_PENDING)
+        {
+            isWaitingForTimer_ = 1;
+            return sleep_and_call(
+                &timer_, SEC_TO_NSEC(3), STATE(write_response_timeout));
+        }
+        else
+        {
+            return call_immediately(STATE(write_response_timeout));
+        }
+    }
+
+    Action write_response_timeout()
+    {
+        if (responseCode_ & DatagramClient::OPERATION_PENDING)
+        {
+            return handle_write_error(Defs::OPENMRN_TIMEOUT);
+        }
+        size_t len = responsePayload_.size();
+        uint8_t *bytes = (uint8_t *)responsePayload_.data();
+        if (len < 6)
+        {
+            LOG(INFO, "Memory Config client: response datagram payload not "
+                      "long enough");
+            return handle_write_error(
+                Defs::ERROR_INVALID_ARGS_MESSAGE_TOO_SHORT);
+        }
+        unsigned ofs;
+        unsigned a = bytes[2];
```

Comment:
Does a refer to address? If so, please replace a with address.

Reply:
done